completions.go

```diff
@@ -1,34 +1,88 @@
 package kingpin
 
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"strings"
+)
+
+type completion struct {
+	flags []string
+	words []string
+}
+
+const (
+	dirCompletionFlag  = "-d"
+	fileCompletionFlag = "-f"
+)
+
 // HintAction is a function type who is expected to return a slice of possible
-// command line arguments.
+// wordlist command line arguments.
```

Comment:
I should've left this unchanged. What I was trying to get at is that these results ultimately will be command line arguments to `compgen`'s wordlist.  But that's not really material to an ordinary API consumer.  Will fix.